### PR TITLE
Filter followon searches from search daily job

### PR DESCRIPTION
**Pull request by harterrt**
I am still figuring out how we can test this on real data without causing a failure when the airflow job runs.

https://bugzilla.mozilla.org/show_bug.cgi?id=1368089

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/python_mozetl/48)

### CODE REVIEW

**Comment by harterrt** [comment]
Just tested this on data from 2017-05-29 with sample_ids in range(10). The data is identical before and after, which is what we hope to see. I don't have any way to test for upcoming followon keys until we refactor this job to be testable. 

Adding Dave as a reviewer to make sure my key filter format makes sense.

**Comment by acmiyaguchi** [comment]
Another +1 for the whitelist approach, its probably more resilient to change. It might be useful to make a comment with a short description of this change and a few examples of keys to be filtered out, as notes to use during a refactor.

File: mozetl/search/daily_search_rollups.py [hunk 1]
```diff
@@ -129,6 +130,7 @@ def roll_up_searches(spark, frame):
         "distribution_id",
         "default_provider",
         "search_counts.engine as search_provider",
+        "search_counts.source as search_source",
```
Comment (acmiyaguchi) [inline mozetl/search/daily_search_rollups.py]: A comment about filtering out search sources not related to search engines. An example of ignored source (`filteron%`).

Comment (harterrt) [inline mozetl/search/daily_search_rollups.py]: See above. The bug links should be the most useful resource for future engineers.

**Comment by harterrt** [comment]
Moved to a whitelist, added a comment, and reran over 3% of one day's worth of main_summary data to verify nothing in the rollup changes.

RFAL.

File: mozetl/search/daily_search_rollups.py [hunk 2]
```diff
@@ -72,6 +77,7 @@ def add_profile_day(frame):
   {}
 WHERE
   ((search_count > -1) OR (search_count is null))
+  AND search_source IN ('searchbar', 'urlbar', 'abouthome', 'newtab', 'contextmenu', 'system')
```
Comment (dzeber) [inline mozetl/search/daily_search_rollups.py]: I think we also need `'NO_SEARCHES'` in here to pull in the rows from the `unwrapped_nulls` table.

Comment (harterrt) [inline mozetl/search/daily_search_rollups.py]: Done, nice catch!

**Comment by harterrt** [comment]
OK - looks like my manual tests passed without good reason. I've added "NO_SEARCHES" to the filter and verified that the version of the code missing this source is broken and the current version matches production.

I'll submit in the morning to make sure nothing new pops up :)

**Comment by harterrt** [comment]
Submitting on Friday and crossing my fingers. We already have to backfill this job for 2017-05-31 and 2017-05-30 for a separate failure, so it will be good to test this job against that data.

**Pull request merged**